web/src/engine/websites/RawLazy.ts

```diff
@@ -0,0 +1,101 @@
+    return text.replace(/\(Raw.*Free\)/i, '').trim();
+}
+
+function ChapterExtractor(this: MangaScraper, anchor: HTMLAnchorElement) {
+    return {
+        id: anchor.pathname,
+        title: anchor.querySelector('span').textContent.trim()
+    };
+}
+
+const findNextStageScript = `
+    new Promise(resolve =>  {
+        const interval = setInterval(function () {
+            if(jQuery('{BUTTONID}').attr('href') != '#'){
+                clearInterval(interval);
+                resolve(jQuery('{BUTTONID}').attr('href'));
+            }
+        }, 500);
+    })
+`;
+
+type APIResult = {
+    mes: string
+    url?: string;
+}
+
+@Common.MangaCSS(/^{origin}\/manga-lazy\/[^/]+\/$/, 'title', MangaLabelExtractor)
+@Common.ChaptersSinglePageCSS('div.chapters-list a', ChapterExtractor)
+@Common.ImageAjax(true)
+export default class extends DecoratableMangaScraper {
+    private nonce: string = undefined;
+
+    public constructor() {
+        super('rawlazy', 'RawLazy', 'https://rawlazy.si', Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Language.Japanese, Tags.Source.Aggregator);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const nonce = await FetchWindowScript<string>(new Request(this.URI), 'zing.nonce');
+        const mangaList = [];
+        for (let page = 1, run = true; run; page++) {
+            const mangas = await this.GetMangasFromPage(page, provider, nonce);
+            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList;
+    }
+
+    private async GetMangasFromPage(page: number, provider: MangaPlugin, nonce: string): Promise<Manga[]> {
+        try {
+
+            const body = new URLSearchParams({
+                action: 'z_do_ajax',
+                _action: 'loadmore',
+                nonce: nonce,
+                p: page.toString(),
+                category_id: '0'
+            }).toString();
+
+            const url = new URL('/wp-admin/admin-ajax.php', this.URI);
+            const request = new Request(url, {
+                credentials: 'include',
+                method: 'POST',
+                body: body.toString(),
+                headers: {
+                    'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
+                    Referer: this.URI.href,
+                    Origin: this.URI.origin,
+                    'X-Requested-With': 'XMLHttpRequest'
+                }
+            });
+
+            const { mes } = await FetchJSON<APIResult>(request);
+            const dom = new DOMParser().parseFromString(mes, 'text/html');
+            const mangas = [...dom.querySelectorAll<HTMLAnchorElement>('div.entry-tag h2 a')];
+            return mangas.map(manga => new Manga(this, provider, manga.pathname, MangaLabelExtractor.call(this, manga)));
+
+        } catch (error) {
+            return [];
```

Comment:
Website dont stop properly and just give error 500.

Reply:
Also, in case we were gone too far, "mes" is not html at all meaning `DOMParser.parsefromString` with trigger and exception. I guess ill have to the the "going" value from results, but error 500 trigger first.

Reply:
Still cant pass rate limit, damn

Reply:
@ronny1982 I keep getting error 500 after a random number of pages with 3 seconds delay. 
Should i just keep that Try-catch?

Reply:
Remove the `try ... catch` and let the error bubble up.
It should be handled by the caller.

Reply:
Piccoma probably has the best fitting example for using a rate limit.
Reaperscan is more complex, but it shows how rate limit can be used in combination with website settings.

Reply:
> Remove the `try ... catch` and let the error bubble up. It should be handled by the caller (the user should keep the 'old' manga list instead of getting it replaced by an incomplete manga list).

Its not possible to get a manga list in the first place since it always end with error 500 but you have a point.

> Piccoma probably has the best fitting example for using a rate limit.

I have no problem applying a rate limit. I am just telling no mater how much i wait between requests, it fails after a while. 2 secondes between requests is already huge and it fails nonetheless.

If more waiting doesnt work I am considering making clipboard only.

Reply:
Website will throw error 🤷